### PR TITLE
[AzureMonitorExporter] Fix DuplicateKeyException when LogRecord contains EventId or EventName in both LogRecord and Attributes

### LINKED ISSUE #44653: [BUG] Azure.Monitor.OpenTelemetry.Exporter EventId duplicate key exception causes LogRecord to be dropped
### Library name and version

Azure.Monitor.OpenTelemetry.Exporter 1.3.0

### Describe the bug

When a LogRecord has an EventId specified and the LogRecord AttributeData dictionary already contains a key EventId then the LogsHelper.GetMessageAndSetProperties method causes a DuplicateKey exception. Which eventually leads to the whole LogRecord getting dropped.

This problem also occurs with attribute key "EventName".

In my project I am using Serilog. Serilog adds (if set on the log entry) EventId and EventName to the log state which causes it to end up in the LogRecord AttributeData dictionary. The LogsHelper then without checking whether it exists adds EventId again to the AttributeData dictionary causing an exception.

https://github.com/Azure/azure-sdk-for-net/blob/Azure.Monitor.OpenTelemetry.Exporter_1.3.0/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/LogsHelper.cs

### Expected behavior

When the attribute dictionary of the LogRecord already contains EventId or EventName then the LogsHelper doesnt try add it again.

### Actual behavior

LogsHelper causes a duplicate key exception and the LogRecord is dropped never gets exported.

### Reproduction Steps

-

### Environment

_No response_

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #44653

If `EventId` and `EventName` are present in both `LogRecord` (`LogRecord.EventId`, `LogRecord.EventName`) and `LogRecord.Attributes`, `EventId` and `EventName` from `LogRecord.Attributes` are honored.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,14 @@
 
 ### Bugs Fixed
 
+* Fixed an issue where a `DuplicateKeyException` could be thrown if `EventId`
+  and `EventName` were present in both `LogRecord` (`LogRecord.EventId`,
+  `LogRecord.EventName`) and `LogRecord.Attributes`. The method now checks if
+  these keys exist before attempting to add them, ensuring that `EventId` and
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:14]: the priority order is not clear.. if attributes and LogRecord both have eventid/name, which one wins?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs [hunk 2]
```diff
@@ -183,6 +183,36 @@ public void PropertiesContainEventIdAndEventNameIfSetOnLog()
...
+                });
+                builder.AddFilter(typeof(LogsHelperTests).FullName, LogLevel.Trace);
+            });
+
+            var logger = loggerFactory.CreateLogger<LogsHelperTests>();
+
+            EventId id = new EventId(1, "TestEvent");
+
+            string log = "Log Information {EventId} {EventName}.";
+            logger.LogInformation(id, log, 100, "TestAttributeEventName");
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs:204]: nit: can you please use named properties here?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs:204]: This is the pattern used in all other tests in this file. If there is no strong objection will leave it unmodified.

**Review by TimothyMothra — APPROVED** [review]

**Comment by vitor-baptista-vfx** [comment]
@rajkumar-rangaraj @TimothyMothra do we have an ETA for releasing this?

**Comment by TimothyMothra** [comment]
@vitor-baptista-vfx We're planning for a beta release the second week of July.

**Pull request merged**